client/gutenberg/editor/utils.js

```diff
@@ -12,17 +12,33 @@ import debugFactory from 'debug';
 
 const debug = debugFactory( 'calypso:gutenberg' );
 
-export const overrideAPIPaths = siteSlug => {
-	// TODO: no API support for now. We'll also need to handle authorization here.
-	apiFetch.use( () => noop );
+export const applyMiddlewares = siteSlug => {
+	//make authenticated calls using the WordPress.com REST Proxy
+	//bypassing the apiFetch call that uses window.fetch
+	//first middleware in, last out
+	apiFetch.use( options => {
+		return proxy(
+			{
+				...options,
+				apiNamespace: 'wp/v2',
+			},
+			( error, body ) => {
+				if ( error ) {
+					return Promise.reject( error );
+				}
+
+				return Promise.resolve( body );
+			}
+		);
+	} );
 
 	const rootURL = 'https://public-api.wordpress.com/';
 	apiFetch.use( apiFetch.createRootURLMiddleware( rootURL ) );
 
 	// rewrite default API paths to match WP.com equivalents
 	// Example: /wp/v2/posts -> /wp/v2/sites/{siteSlug}/posts
 	apiFetch.use( ( options, next ) => {
-		const wpcomPath = `/wp/v2/sites/${ siteSlug }/` + options.path.replace( '/wp/v2/', '' );
+		const wpcomPath = `/sites/${ siteSlug }` + options.path.replace( '/wp/v2/', '/' );
```

Comment:
We could probably shorten this to:
```javascript
const wpcomPath = options.path.replace( '/wp/v2/', `/sites/${siteSlug}/` );
```

Reply:
Very wise. Updated on fca7b6135a91b47113af39793ebf0812a14bb32b.